neoscore/western/chordrest.py

```diff
@@ -333,47 +337,51 @@ def extra_attachment_point(self) -> Point:
         x = notehead.x + bounding_rect.x + (bounding_rect.width / 2)
         return Point(x, y)
 
+    @cached_property
     def tremolo_attachment_point(self) -> Point:
-        """A point where tremolo's attach.
+        """A convenient reasonable location for tremolos to be placed.
 
         For chords, this is a point centered above or below the outermost notehead
-        opposite of the stem direction.
-
-        For rests, this is a point centered above the rest.
+        opposite of the stem direction. For rests, this behavior is undefined.
 
         The returned point is relative to the Chordrest.
         """
         if self.rest:
-            bounding_rect = self.rest.bounding_rect
-            x = self.rest.x + bounding_rect.x + (bounding_rect.width / 2)
-            y = self.rest.y + bounding_rect.y - self.staff.unit(1)
-            return Point(x, y)
+            # Since there's no use-case we know of for rest tremolos, there's no
+            # reasonable implementation of this, so just return 0, 0.
+            return ORIGIN
+        x = ZERO
+        noteheads_rect = self.notehead_column_bounding_rect
+        if not self.duration.display.requires_stem:
+            x = noteheads_rect.x + (noteheads_rect.width / 2)
         if self.stem_direction == DirectionY.DOWN:
-            notehead = self.lowest_notehead
-            bounding_rect = notehead.bounding_rect
-            x = notehead.x - bounding_rect.x
-            y = (
-                notehead.y
-                + bounding_rect.y
-                + bounding_rect.height
-                + self.staff.unit(1.5)
-            )
+            y = noteheads_rect.y + noteheads_rect.height + self.staff.unit(1.5)
         else:
-            notehead = self.highest_notehead
-            bounding_rect = notehead.bounding_rect
-            x = notehead.x + bounding_rect.x + bounding_rect.width
-            y = notehead.y - bounding_rect.y - bounding_rect.height - self.staff.unit(1)
+            y = noteheads_rect.y - self.staff.unit(1)
```

Comment:
btw - how'd you come up with the offset difference for above vs below?

Reply:
also, check out the example. the positioning looks pretty tight for the two down-stem cases. it's beyond the scope of this PR, but we probably need to support stem height overrides for this kind of thing. chordrest today has no ability to change the stem height (beside manually going in after init and altering the stem component)

Reply:
For the offset difference I basically copied your extra adjustment code and split in half. It was a lucky guess, and I think there could be a more intelligent version especially if a 1/4 note or 1/8 needs 5 stroke tremolos (a legit ask especially if tempos are low).

very happy with your changes. Please commit and merge. And accept my sincere thanks.